### PR TITLE
README.md and CHANGELOG.md updates

### PR DESCRIPTION
**Pull request by mikewiebe**
Updates to add new cisco and netdev resources.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1095,6 +1128,192 @@ VTP file name. Valid values are a string or the keyword 'default'.
...
+Contact name for this device.  Valid value is a string.
+
+##### `location`
+Location of this device.  Valid value is a string.
+
+### Type: ntp_config
+
+#### Parameters
+
+##### `ensure - Check this one.  Really ensureable?`
```
Comment (mikewiebe) [inline README.md]: All of these "Really ensureable" lines will be updated after I test them in the demo manifest.

File: README.md [hunk 2]
```diff
@@ -154,21 +154,40 @@ cisco_interface_ospf {"Ethernet1/2 Sample":
...
 * SNMP Types
   * [`cisco_snmp_community`](#type-cisco_snmp_community)
   * [`cisco_snmp_group`](#type-cisco_snmp_group)
   * [`cisco_snmp_server`](#type-cisco_snmp_server)
   * [`cisco_snmp_user`](#type-cisco_snmp_user)
+  * [`network_snmp (netdev_stdlib)`](#type-network_snmp)
+
+* SYSLOG Types (netdev_stdlib)
+  * [`syslog_server`](#type-syslog_server)
+  * [`syslog_settings`](#type-syslog_settings)
```
Comment (chrisvanheuveln) [inline README.md]: I'd prefer to see this as:

```
* SYSLOG Types
  * [`syslog_server (netdev_stdlib)`](#type-syslog_server)
  * [`syslog_settings (netdev_stdlib)`](#type-syslog_settings)
```

File: README.md [hunk 3]
```diff
@@ -1095,6 +1128,192 @@ VTP file name. Valid values are a string or the keyword 'default'.
...
+### Type: network_dns
+
+#### Parameters
+
+##### `ensure`
+Determines whether or not the config should be present on the device. Valid
+values are 'present' and 'absent'. Default value is 'present'.
+
+##### `name`
+Name, generally "settings", not used to manage the resource.  Valid value is a string.
```
Comment (chrisvanheuveln) [inline README.md]: I'm not sure what this is trying to say exactly...  Name not used to manage the resource?

File: README.md [hunk 4]
```diff
@@ -1095,6 +1128,192 @@ VTP file name. Valid values are a string or the keyword 'default'.
...
+
+##### `domain`
+Default domain name to append to the device hostname.  Valid value is a string.
+
+##### `search`
+Array of DNS suffixes to search for FQDN entries.  Valid value is an array of strings.
```
Comment (chrisvanheuveln) [inline README.md]: I think it would be good to include a short example of the array. The array types are always confusing.

File: README.md (hunk 3 above)
Comment (mikewiebe) [inline README.md]: Not sure either.. this is verbatim verbage from the netdev type.

File: README.md [hunk 5]
```diff
@@ -1095,6 +1128,192 @@ VTP file name. Valid values are a string or the keyword 'default'.
...
+
+##### `servers`
+Array of DNS servers to use for name resolution.  Valid value is an array of strings.
+
+### Type: network_snmp
+
+#### Parameters
+
+##### `name`
+Name of the Puppet resource, not used to manage the device.  Valid value is a string.
```
Comment (chrisvanheuveln) [inline README.md]: Could be clearer.

File: README.md [hunk 6]
```diff
@@ -1095,6 +1128,192 @@ VTP file name. Valid values are a string or the keyword 'default'.
...
+
+### Type: ntp_config
+
+#### Parameters
+
+##### `ensure - Check this one.  Really ensureable?`
+Determines whether or not the config should be present on the device. Valid values are 'present' and 'absent'. Default value is 'present'.
+
+##### `name`
+Resource name, not used to configure the device.  Valid value is a string.
```
Comment (chrisvanheuveln) [inline README.md]: ditto

File: README.md (hunk 4 above)
Comment (mikewiebe) [inline README.md]: Agreed.. I would prefer to commit this and then I will update after i test this stuff using the demo manifest

File: README.md (hunk 5 above)
Comment (mikewiebe) [inline README.md]: NetDev verbage.. was trying to be consistent with that.

File: README.md [hunk 7]
```diff
@@ -1095,6 +1130,192 @@ VTP file name. Valid values are a string or the keyword 'default'.
...
+
+##### `source_interface`
+Source interface for the NTP system.  Valid value is a string.
```
Comment (chrisvanheuveln) [inline README.md]: "system" or "server"?

File: README.md (hunk 6 above)
Comment (mikewiebe) [inline README.md]: Same comment as above...

File: README.md [hunk 8]
```diff
@@ -1095,6 +1130,192 @@ VTP file name. Valid values are a string or the keyword 'default'.
...
+#### Parameters
+
+##### `ensure - Check this one.  Really ensureable?`
+Determines whether or not the config should be present on the device. Valid values are 'present' and 'absent'. Default value is 'present'.
+
+##### `name`
+Hostname or address of the Syslog server.  Valid value is a string.
+
+##### `time_stamp_units`
+The unit to log time values in.  Valid values are 'seconds' and 'milliseconds'.
```
Comment (chrisvanheuveln) [inline README.md]: `/The unit to log time values in. /The unit of measurement for log time values./`

File: README.md (hunk 7 above)
Comment (mikewiebe) [inline README.md]: From the type....

```
newproperty(:source_interface) do
    desc 'The source interface for the NTP system'

    validate do |value|
      if value.is_a? String then super(value)
      else fail "value #{value.inspect} is invalid, must be a String."
      end
    end
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -2,14 +2,28 @@
...
+- domain_name provider.
+- name_server provider.
+- network_dns provider.
+- network_snmp provider.
+- ntp_config provider.
+- ntp_server provider.
+- radius provider.
+- radius global provider.
+- syslog_server provider.
+- syslog_setting provider.
```
Comment (chrisvanheuveln) [inline CHANGELOG.md]: Nicely alphabetized. :-)

File: README.md (hunk 7 above)
Comment (chrisvanheuveln) [inline README.md]: Use server.

File: CHANGELOG.md (hunk 9 above)
Comment (mikewiebe) [inline CHANGELOG.md]: Work I extra to hard alphabetize.

File: README.md (hunk 7 above)
Comment (mikewiebe) [inline README.md]: Fixed

File: README.md (hunk 8 above)
Comment (mikewiebe) [inline README.md]: Fixed

**Pull request merged**